js/src/offcanvasmenu.js

```diff
@@ -0,0 +1,393 @@
+    this._parent          = null
+    this._element         = null
+    this._triggerArray    = null
+    this._isTransitioning = null
+  }
+
+  // Private
+
+  _getConfig(config) {
+    config = {
+      ...Default,
+      ...config
+    }
+    config.toggle = Boolean(config.toggle) // Coerce string values
+    Util.typeCheckConfig(NAME, config, DefaultType)
+    return config
+  }
+
+  _getParent() {
+    let parent
+
+    if (Util.isElement(this._config.parent)) {
+      parent = this._config.parent
+
+      // It's a jQuery object
+      if (typeof this._config.parent.jquery !== 'undefined') {
+        parent = this._config.parent[0]
+      }
+    } else {
+      parent = document.querySelector(this._config.parent)
+    }
+
+    const selector = `[data-toggle="offcanvas"][data-parent="${this._config.parent}"]`
+    const children = [].slice.call(parent.querySelectorAll(selector))
+
+    $(children).each((i, element) => {
+      this._addAriaAndOffcanvasmenudClass(
+        Offcanvasmenu._getTargetFromElement(element),
+        [element]
+      )
+    })
+
+    return parent
+  }
+
+  _addAriaAndOffcanvasmenudClass(element, triggerArray) {
+    const isOpen = $(element).hasClass(CLASS_NAME_OPEN)
+
+    if (triggerArray.length) {
+      $(triggerArray)
+        .toggleClass(CLASS_NAME_CLOSED, !isOpen)
+        .attr('aria-expanded', isOpen)
+    }
+  }
+
+  // Static
+
+  static _getTargetFromElement(element) {
+    const selector = Util.getSelectorFromElement(element)
+    return selector ? document.querySelector(selector) : null
+  }
+
+  static _jQueryInterface(config) {
+    return this.each(function () {
+      const $this   = $(this)
+      let data      = $this.data(DATA_KEY)
+      const _config = {
+        ...Default,
+        ...$this.data(),
+        ...typeof config === 'object' && config ? config : {}
+      }
+
+      if (!data && _config.toggle && typeof config === 'string' && /open|close/.test(config)) {
+        _config.toggle = false
+      }
+
+      if (!data) {
+        data = new Offcanvasmenu(this, _config)
+        $this.data(DATA_KEY, data)
+      }
+
+      if (typeof config === 'string') {
+        if (typeof data[config] === 'undefined') {
+          throw new TypeError(`No method named "${config}"`)
+        }
+        data[config]()
+      }
+    })
+  }
+}
+
+/**
+ * ------------------------------------------------------------------------
+ * Viewport conditional dropdown menu override for offcanvas meun.
+ * ------------------------------------------------------------------------
+ */
+
+// Define our viewportWidth variable
+let VIEWPORT_WIDTH = false
+const XS_BREAKPOINT_MAX = 767
+
+// @TODO Use CSS breakpoint info, rather than seemingly arbitrary window width.
+// Set/update the viewportWidth value
+function setViewportWidth() {
+  VIEWPORT_WIDTH = window.innerWidth || document.documentElement.clientWidth
+}
+
+$('.dropdown.keep-open .dropdown-toggle').on('click', function (event) {
```

Comment:
Should we target `.navbar-offcanvas .dropdown` instead of all `.dropdown` elements? Since there can be `.dropdown`s used outside of the main navigation.